lib/logqueue-fifo.c

```diff
@@ -539,8 +547,8 @@ log_queue_fifo_new(gint qoverflow_size, const gchar *persist_name)
     {
       INIT_IV_LIST_HEAD(&self->qoverflow_input[i].items);
       worker_batch_callback_init(&self->qoverflow_input[i].cb);
-      self->qoverflow_input[i].cb.user_data = self;
       self->qoverflow_input[i].cb.func = log_queue_fifo_move_input;
+      self->qoverflow_input[thread_id].cb.user_data = self;
     }
```

Comment:
I think there's a mistake here. If you look at the earlier initialization that was removed, it uses "i" as index, whereas this uses thread_id, which means that not all user_data slots will be initialized. Actually, I can't even find thread_id as a variable declared, but maybe it is a global from somewhere.